sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/NamedVIPSpec.java

```diff
@@ -80,4 +94,48 @@ public boolean equals(Object o) {
     public int hashCode() {
         return HashCodeBuilder.reflectionHashCode(this);
     }
+
+    /**
+     * {@link NamedVIPSpec} builder static inner class.
+     */
+    public static class Builder extends PortSpec.Builder {
+
+        private String protocol;
+        private String vipName;
+        private Integer vipPort;
+
+        private Builder() {
+        }
+
+        public Builder protocol(String protocol) {
+            this.protocol = protocol;
+            return this;
+        }
+
+        public Builder vipName(String vipName) {
+            this.vipName = vipName;
+            return this;
+        }
+
+        public Builder vipPort(Integer vipPort) {
+            this.vipPort = vipPort;
+            return this;
+        }
+
+        public NamedVIPSpec build() {
+            PortSpec portSpec = super.build();
+            return new NamedVIPSpec(
```

Comment:
This means we don't call the builder constructor, correct? Does this mean that we miss the validation?

Reply:
Yes, it got weird with these inherited types. I broke out the validation into separate functions to be explicitly called in the subtypes.